### PR TITLE
all: Call runtime.GOMAXPROCS(0) once on startup.

### PR DESCRIPTION
**Pull request by shurcooL**
Instead of calling it once per request.

Its current performance is not as bad as I might've imagined, but it
still seems scary to call a runtime method that often. Its value
doesn't change mid-process, so might as well store it. This way,
accessing the value does not need any locks.

### CODE REVIEW

**Comment by shurcooL** [comment]
Hmm, I just realized this might be invalid if any code actually calls `runtime.GOMAXPROCS` with a > 1 value and *changes* its value, since `conf.GoMaxProcs` would be out of date then.

Perhaps we should be using `runtime.NumCPU` instead?

My goal was to avoid calling sometime from `runtime` on a per-request basis.

**Comment by shurcooL** [comment]
Closing as invalid.

> My goal was to avoid calling sometime from `runtime` package on a per-request basis.

I'll try a different take on that, just use reasonable constants.

File: cli/internal/middleware/status.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func statusHandler(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintln(w, "Hostname:   ", hostname)
 	fmt.Fprintln(w, "Git commit: ", envutil.GitCommitID)
 	fmt.Fprintln(w, "Uptime:     ", time.Since(sgxStarted))
-	fmt.Fprintln(w, "GOMAXPROCS: ", runtime.GOMAXPROCS(0))
```
Comment (keegancsmith) [inline cli/internal/middleware/status.go]: @shurcooL I'd vote to just remove this from the status handler. It's information we don't actually care about.

Comment (shurcooL) [inline cli/internal/middleware/status.go]: If it's not needed, I have no problem with removing it.

Made https://github.com/sourcegraph/sourcegraph/pull/246.

**Pull request closed without merging**